### ISSUE TITLE
python 3 support?

### ISSUE THREAD

**Issue by ajwillo** [O365/python-o365#63]
Hi,

my code sample below:

```
#Get Mail
    i = Inbox(settings.ON_CALL_MAILBOX_AUTH,getNow=False)
    # filter sample (IsRead eq false) and (Subject eq 'who')
    i.getMessages()

    messages = []
    for mail in i.messages:  
        obj_mail = Mail(
            sender = str(Message.getSender(mail)['EmailAddress']['Address']),
            subject = Message.getSubject(mail),
            content = Message.getBody(mail)
        )
        messages.append(obj_mail)
```

is returning an error
```
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/O365/inbox.py", line 56, in getMessages
    for message in response.json()['value']:
  File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```

im assuming this is a python 3 issue? is there a fork of a python 3 version anywhere?

Thanks

**Comment by Narcolapser** [comment]
Hmm, strange. No, there is no separate fork for py3, I've tried to keep the library compatible with both py2 and py3. You may be right on this being a problem that exists in py3 and not in py2 though. Have you tried running it with py2? 

What I'm wonderning about now is what exactly is the json response that it is causing problems. Run the code through a debugger and see if you can get the content of response.json().

**Comment by drewstinnett** [comment]
I hit this error as well in py3, it ended up just being that I hadn't put the right credentials in.  If you tack this on to the top of your script, do you get a 401 response?

```python
    import logging
    logging.basicConfig(level=logging.DEBUG)
```

**Comment by Narcolapser** [comment]
hey @drewstinnett can you post your code? If this is a problem for several people and you've found a work around I'd like to get that included in the library.

**Comment by drewstinnett** [comment]
For sure, here's what I see:

```python
#!/usr/bin/env python3

from O365 import Connection, FluentInbox
import logging

logging.basicConfig(level=logging.DEBUG)

Connection.login('bad-username', 'bad-password')
inbox = FluentInbox()

messages = inbox.search('Subject:foo').fetch(count=20)
print(messages)
```

This is what I get when running it (note the 401 in the second DEBUG line)

```
 $ python3 ./test_office_py3.py
INFO:O365.connection:Requesting URL: https://outlook.office365.com/api/v1.0/me/messages
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): outlook.office365.com
DEBUG:urllib3.connectionpool:https://outlook.office365.com:443 "GET /api/v1.0/me/messages?%24filter=&%24top=20&%24search=%22Subject%3Afoo%22 HTTP/1.1" 401 0
INFO:O365.connection:Received response from URL https://outlook.office365.com/api/v1.0/me/messages?%24filter=&%24top=20&%24search=%22Subject%3Afoo%22
Traceback (most recent call last):
  File "./test_office_py3.py", line 11, in <module>
    messages = inbox.search('Subject:foo').fetch(count=20)
  File "/usr/local/lib/python3.6/site-packages/O365/fluent_inbox.py", line 183, in fetch
    return self.fetch_next(count=count)
  File "/usr/local/lib/python3.6/site-packages/O365/fluent_inbox.py", line 199, in fetch_next
    params=params)
  File "/usr/local/lib/python3.6/site-packages/O365/connection.py", line 220, in get_response
    response_json = response.json()
  File "/usr/local/lib/python3.6/site-packages/requests/models.py", line 892, in json
    return complexjson.loads(self.text, **kwargs)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python3/3.6.4_2/Frameworks/Python.framework/Versions/3.6/lib/python3.6/json/decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

I actually get a really similar error when using python2, so I don't think it's necessarily a v3 thing

**Comment by drewstinnett** [comment]
Interesting...this is what I get when using a valid username/password:

```
INFO:O365.connection:Requesting URL: https://outlook.office365.com/api/v1.0/me/messages
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): outlook.office365.com
DEBUG:urllib3.connectionpool:https://outlook.office365.com:443 "GET /api/v1.0/me/messages?%24filter=&%24top=20&%24search=%22Subject%3Afoo%22 HTTP/1.1" 200 None
INFO:O365.connection:Received response from URL https://outlook.office365.com/api/v1.0/me/messages?%24filter=&%24top=20&%24search=%22Subject%3Afoo%22
[<O365.fluent_message.Message object at 0x105bc2898>, <O365.fluent_message.Message object at 0x1065fefd0>, <O365.fluent_message.Message object at 0x107361978>, <O365.fluent_message.Message object at 0x1073619b0>, <O365.fluent_message.Message object at 0x1073619e8>, <O365.fluent_message.Message object at 0x107361940>, <O365.fluent_message.Message object at 0x107361c50>, <O365.fluent_message.Message object at 0x107361c18>, <O365.fluent_message.Message object at 0x107361be0>, <O365.fluent_message.Message object at 0x107361d68>, <O365.fluent_message.Message object at 0x107361e80>, <O365.fluent_message.Message object at 0x107361cf8>, <O365.fluent_message.Message object at 0x107361d30>, <O365.fluent_message.Message object at 0x106c75d30>, <O365.fluent_message.Message object at 0x10737f828>, <O365.fluent_message.Message object at 0x10737f5f8>, <O365.fluent_message.Message object at 0x10737f8d0>, <O365.fluent_message.Message object at 0x10737f710>, <O365.fluent_message.Message object at 0x10737f630>, <O365.fluent_message.Message object at 0x10737f748>]
```

**Comment by Narcolapser** [comment]
So it looks like it works fine in that case. So what we are dealing with then is an edge case where it breaks when there is no messages in the responses. Same idea as the problem ajwillo had but different part of the library. Hey @roycem90 fluent is kinda your baby, want to take a look at this?

**Comment by roycem90** [comment]
Closing this too.. as the new api is completly built on python 3

and the no response case is already fixed on the python-2.7 branch
